tests/rest/admin/test_event_reports.py

```diff
@@ -378,6 +378,37 @@ def test_next_token(self) -> None:
         self.assertEqual(len(channel.json_body["event_reports"]), 1)
         self.assertNotIn("next_token", channel.json_body)
 
+    def test_filter_against_user(self) -> None:
+        # first grab all the reports
+        channel = self.make_request(
+            "GET",
+            self.url,
+            access_token=self.admin_user_tok,
+        )
+        self.assertEqual(channel.code, 200)
+
+        # filter out set of report ids of events sent by one of the users
+        filtered_report_ids = []
+        for event_report in channel.json_body["event_reports"]:
+            if event_report["sender"] == self.other_user:
+                filtered_report_ids.append(event_report["id"])
+
+        # grab the report ids by sender and compare to filtered report ids
+        channel = self.make_request(
+            "GET",
+            f"{self.url}?sender_user_id={self.other_user}",
+            access_token=self.admin_user_tok,
+        )
+        self.assertEqual(200, channel.code)
+        self.assertEqual(channel.json_body["total"], len(filtered_report_ids))
+
+        event_reports = channel.json_body["event_reports"]
+        returned_report_ids = []
+        for event_report in event_reports:
+            if event_report["sender"] == self.other_user:
+                returned_report_ids.append(event_report["id"])
```

Comment:
Might as well make a `Set` from the start

```suggestion
        returned_report_ids = set()
        for event_report in event_reports:
            if event_report["sender"] == self.other_user:
                returned_report_ids.add(event_report["id"])
```